src/server/game/Conditions/ConditionMgr.h

```diff
@@ -286,6 +288,7 @@ class TC_GAME_API ConditionMgr
         bool IsObjectMeetingVehicleSpellConditions(uint32 creatureId, uint32 spellId, Player* player, Unit* vehicle) const;
         bool IsObjectMeetingSmartEventConditions(int64 entryOrGuid, uint32 eventId, uint32 sourceType, Unit* unit, WorldObject* baseObject) const;
         bool IsObjectMeetingVendorItemConditions(uint32 creatureId, uint32 itemId, Player* player, Creature* vendor) const;
+        bool IsObjectMeetingAreaTriggerConditions(std::pair<uint32, bool> const& areaTriggerId, WorldObject* object) const;
```

Comment:
personally i kinda dont like std::pair here since we could use AreaTriggerId directly, but that either requires us to include AreaTriggerTemplate.h in ConditionMgr (which feels way more worse) or move AreaTriggerId to a new file, which feels weird. Or we could use a pointer and just deref it inside + forward declare in ConditionMgr.h, but that also feels weird.

Reply:
But the rule to allow forward declaring is not "use pointer", its "don't use value"